app/src/main/java/com/babylon/wallet/android/presentation/dialogs/dapp/DAppDetailsDialogViewModel.kt

```diff
@@ -46,6 +61,34 @@ class DAppDetailsDialogViewModel @Inject constructor(
                 }
             }
         }
+        observeDapp()
+    }
+
+    fun onDeleteDapp() {
+        viewModelScope.launch {
+            dAppConnectionRepository.deleteAuthorizedDApp(args.dAppDefinitionAddress)
+            sendEvent(Event.DAppDeleted)
+        }
+    }
+
+    private fun observeDapp() {
+        viewModelScope.launch {
+            dAppConnectionRepository.getAuthorizedDAppFlow(args.dAppDefinitionAddress).collect {
```

Comment:
do we really need the flow here? I mean you can use the suspend `getAuthorizedDApp` for this screen, right?
Unless someone takes a login request on top of the dialog and a new persona is used. 😄 
Then I think we indeed need the flow.

Reply:
Didn't test that case with request handled on top of the dialog, but case you described seem valid, dApp can be updated like this